src/modules/universal-discovery/components/content-creator/content.creator.component.js

```diff
@@ -24,7 +24,15 @@ export default class ContentCreatorComponent extends Component {
 
     handlePublish() {
         this._refIframe.contentWindow.onbeforeunload = () => {};
-        this._refIframe.contentWindow.document.body.querySelector('#ezrepoforms_content_edit_publish').click();
+        this.submitForm();
+    }
+
+    submitForm() {
+        const submit = this._refIframe.contentWindow.document.body.querySelector('[type="submit"]');
```

Comment:
This will send the first form which has a button type submit. As I suggested we need custom data-attribute.

Reply:
yes, because by design it can be only one form with just one submit button and this is guaranteed by the backend. Adding additional special selector just worsen developer experience.

Reply:
No, it's not. This is fully extensible.

Reply:
I agree with @dew326 This is too generic. We need to be precise which form are going to submit.

Reply:
@pawbuj Form can have multiple submit buttons and often does, especially regarding content creation. This form is fully extensible and different submit values are used to generate different events (used by FormProcessors in RepositoryFormsBundle).

Example ref. https://github.com/ezsystems/repository-forms/blob/master/bundle/Controller/ContentEditController.php#L114

Reply:
@Nattfarinn You misunderstand me, I am not saying in general of course :) but about this particular case: "creating content via on the fly modal". We use here just 2 forms, one in the same time 
https://github.com/ezsystems/ezplatform-admin-ui/blob/554d06c0a5b3d381c2d6a2485b0bd5b29654bbfe/src/bundle/Controller/ContentOnTheFlyController.php#L107
https://github.com/ezsystems/ezplatform-admin-ui/blob/EZP-29094/src/bundle/Controller/ContentOnTheFlyController.php#L143

each of those forms has just one submit button 
https://github.com/ezsystems/ezplatform-admin-ui/blob/554d06c0a5b3d381c2d6a2485b0bd5b29654bbfe/src/bundle/Resources/views/content/content_on_the_fly/content_create_on_the_fly.html.twig#L30-L36
and this is guaranteed by the backend.

To finish the discussion, I will add an additional selector.

Reply:
> We use here just 2 forms, one in the same time (...) each of those forms has just one submit button (...) and this is guaranteed by the backend.

I am not sure what you mean by "each of those forms has just one submit button (...) and this is guaranteed by the backend". Each of those forms **have** *multiple* hidden submit buttons. Even with `drafts_enabled: false` there is also FlexWorkflowBundle, DateBasedPublisherBundle, preview and workflow. Each of these features adds at least one submit button to `ContentEditType` which is used in Content on the Fly as well.

You should never rely on *get me first submit button* if there is a way to add/remove/rearrange buttons as you like. You are missing the point we provide fully extensible software while we keep a Backward Compatibility on our features. If it is **extensible** it is **not** guaranteed by the backend.